lambdas/service/app.py

```diff
@@ -758,57 +758,69 @@ def repository_search(entity_type: str, item_id: str):
 
 
 def filters_param_spec(facets):
+    array_pair_spec = schema.array({}, minItems=2, maxItems=2)
+
+    filter_schema = schema.object(**{
+        facet: schema.optional({
+            'oneOf': [
+                schema.object(**{'is': schema.array({})}),
+                *[
+                    schema.object(**{op: array_pair_spec})
+                    for op in ['contains', 'within', 'intersects']
+                ]
+            ]
+        })
+        for facet in facets
+    })
+    filter_schema['example'] = {'cellCount': {'within': [[10000, 1000000000]]}}
+    # Renders as "OrderedMap {}" in Swagger UI 3.24.2 unless passed as a string
+    filter_schema['default'] = "{}"
     return params.query(
         'filters',
-        schema.optional(
-            application_json(
-                schema.object(**{
-                    facet: schema.optional(
-                        schema.object(**{
-                            relation: schema.optional(schema.array({}))
-                            for relation in ('is', 'contains', 'within', 'intersects')
-                        })
-                    ) for facet in facets
-                }),
-            )
-        ),
+        schema.optional(application_json(filter_schema)),
         description=format_description('''
-            Criteria to filter entities from the search results. Omit
-            parameter or use `{}` to skip filtering. Each filter consists of a
-            facet, a relationship, and an array of relationship parameters. For
-            the "is" relationship, the parameters are simple values and multiple
-            parameters are combined using "or" logic. For example,
-            `{"is":["foo", "bar"]}` will check for equality with either "foo" or
-            "bar". For the other relationships, the parameters are pairs
-            specifying upper and lower bounds, and multiple parameters are
-            combined using "and" logic. For example,
-            `{"within":[[1, 20], [10, 50]]}` will check for falling within both of
-            the specified ranges, i.e. between 10 and 20.
-        '''))
+            Criteria to filter entities from the search results.
+
+            Each filter consists of a facet, an operator, and an array of facet
```

Comment:
https://github.com/DataBiosphere/azul/pull/1647#discussion_r410340722

Please make sure my change requests are honoured either by applying the requested changes or by replying with a comment. If I see this occurring more frequently, my reviews are going to get very, very focused and there are going to be many more cycles.

Reply:
Please address ALL my previous comments on this `description` section.